ships/0031-shipwright-trigger.md

```diff
@@ -0,0 +1,296 @@
+
+### Tekton Pipelines
+
+Let's zoom in on an example of triggering the Shipwright Build when a Tekton Pipeline named "nodejs-ex" has succeeded. For instance:
+
+```yaml
+---
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+spec:
+  trigger:
+    when:
+      - name: nodejs-ex pipeline has succeeded
+        type: Pipeline
+        objectRef:
+          name: nodejs-ex
+          status:
+            - Succeeded
+```
+
+This gives Shipwright the ability to build a container image when the continuous-integration pipeline has ended, like for instance building an image after the tests have finished. Another example is watching over dependencies and other automation processes running on Tekton.
+
+The API also supports using selector labels to match Tekton Pipelines as well, giving a broader scope to select Pipelines to trigger Builds.
+
+To accomplish this task, Shipwright Trigger must watch over PipelineRun objects, and check if there are Builds in the inventory that should be activated. The business logic is recorded on the [PipelineRun Controller](https://github.com/otaviof/shipwright-trigger/blob/main/pkg/trigger/controllers/pipelinerun_controller.go).
+
+## Trigger Actors
+
+This section explains what are the actions inside the Shipwright Trigger application, and how they interact with each other and with Kubernetes as well. Please consider the following image:
+
+![Shipwright Trigger Components](assets/0031-shipwright-trigger-components.drawio.png)
+
+In the next sections each actor is detailed.
+
+### Build Inventory
+
+The [inventory is the central component](https://github.com/otaviof/shipwright-trigger/blob/main/pkg/trigger/inventory/inventory.go) of Shipwright Trigger, it stores all the Build instances organized in a way that allows searching for types of triggers, depending on the Inventory client.
+
+For example, the WebHook Handler will always search for Builds based on the Git repository URL, the type of event (`Push` or `PullRequest`), and the branch names. In other hand, the other Controllers will query the inventory based on the `.objectRef` attribute instead.
+
+As you can see on the diagram above, almost all components are interacting with the Inventory using the specialized query methods `SearchForGit` and `SearchForObjectRef`.
+
+#### Shipwright Build Controller
+
+The Builds are added or removed from the Inventory through the [Build Controller](https://github.com/otaviof/shipwright-trigger/blob/main/pkg/trigger/controllers/build_controller.go), responsible to reflect all Shipwright Build resources into the Inventory. On adding new entries, the Build is prepared for the subsequent queries.
```

Comment:
As mentioned earlier, I do not think that the term controller is suitable here because we do not do anything (beside caching) when some Build changes. On the other hand, Tekton Run is really a controller because we do something (setup BuildRun, update Run status).

Reply:
I'm borrowing the term "controller" from Kubernetes, as in a actor which watches over a resource moving it to the desired state. In this case the so called "Build Controller" is watching over `Builds` and feeding the Inventory... what do you think would be a more suitable name to it?

/cc @adambkaplan @gabemontero